docs/source/format/Columnar.rst

```diff
@@ -765,6 +765,66 @@ application.
 We discuss dictionary encoding as it relates to serialization further
 below.
 
+.. _run-length-encoded-layout:
+
+Run-Length Encoded Layout
+-------------------------
+
+Run-Length is a data representation that represents data as sequences of the
+same value, called runs. Each run is represented as a value, and an integer
+describing how often this value is repeated.
+
+Any array can be run-length encoded. A run-length encoded array has no buffers
+by itself, but has two child arrays. The first one holds a signed 32-bit integer
+for each run. The actual values of each run are held the second child array.
+
+The values in the first child array represent the length of each run. They do
+not hold the length of the respective run directly, but the accumulated length
+of all runs from the first to the current one, i.e. the logical index where the
+current run ends. This allows relatively efficient random access from a logical
+index using binary search. The length of an individual run can be determined by
+subtracting two adjacent values.
+
+A run has to have a length of at least 1. This means the values in the
+run ends array all positive and in strictly ascending order. A run end cannot be
+null.
```

Comment:
I recommend adding something to this section saying "RLE Arrays must not have have a validity bitmap" which would also imply a run end could not be null. As worded I think it is unclear if a validity bitmap is allowed but must be all `valid` or if the bitmap is not allowed

Reply:
" A run end cannot be null." sounds good to me